src/rlookup.c

```diff
@@ -519,50 +520,19 @@ RSValue* jsonIterToValueExpanded(RedisModuleCtx *ctx, JSONResultsIterator iter)
 //
 // Multi value is supported with apiVersion >= APIVERSION_RETURN_MULTI_CMP_FIRST
 int jsonIterToValue(RedisModuleCtx *ctx, JSONResultsIterator iter, unsigned int apiVersion, RSValue **rsv) {
```

Comment:
consider adding a comment telling that this function consumes (or taking ownership) on the iterator if it succeed